File: project-code/cs50x/problem-set-1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int cents;

    // Prompt the user for the amount of change owed, re-prompt if input is invalid
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    // Initialize coin counters and available denominations
    int coins = 0;
    int quarters = 25;
    int dimes = 10;
    int nickels = 5;
    int pennies = 1;

    // Calculate the minimum number of coins
    while (cents >= quarters)
    {
        cents -= quarters;
        coins++;
    }
    while (cents >= dimes)
    {
        cents -= dimes;
        coins++;
    }
    while (cents >= nickels)
    {
        cents -= nickels;
        coins++;
    }
    while (cents >= pennies)
    {
        cents -= pennies;
        coins++;
    }

    // Print the total number of coins
    printf("%d\n", coins);

    return 0;
}

File: cs50x/problem-set-6/sentimental-cash/cash.py
from cs50 import get_float


def main():
    while True:
        change_owed = get_float("Change owed: ")
        if change_owed >= 0:
            break

    # Convert dollars to cents to avoid floating-point precision issues
    cents = round(change_owed * 100)

    coins = 0

    # Calculate the number of quarters
    coins += cents // 25
    cents %= 25

    # Calculate the number of dimes
    coins += cents // 10
    cents %= 10

    # Calculate the number of nickels
    coins += cents // 5
    cents %= 5

    # Calculate the number of pennies
    coins += cents

    print(coins)


if __name__ == "__main__":
    main()
